tombstone_gc.hh

```diff
@@ -28,23 +30,38 @@ class database;
 
 }
 
-class repair_history_map;
-using per_table_history_maps = std::unordered_map<table_id, seastar::lw_shared_ptr<repair_history_map>>;
+using reconcile_history_map = boost::icl::interval_map<dht::token, gc_clock::time_point, boost::icl::partial_absorber, std::less, boost::icl::inplace_max>;
+
+class per_table_history_maps {
+public:
+    std::unordered_map<table_id, seastar::lw_shared_ptr<reconcile_history_map>> _repair_maps;
+
+    // separating the group0 map - it is not kept per table, but for the whole group0
+    seastar::lw_shared_ptr<reconcile_history_map> _group0_map;
+};
 
 class tombstone_gc_options;
 
 using gc_time_min_source = std::function<gc_clock::time_point(const table_id&)>;
 
 class tombstone_gc_state {
     gc_time_min_source _gc_min_source;
-    per_table_history_maps* _repair_history_maps;
+    per_table_history_maps* _reconcile_history_maps;
     gc_clock::time_point check_min(schema_ptr, gc_clock::time_point) const;
+
+    seastar::lw_shared_ptr<reconcile_history_map> get_repair_history_for_table(const table_id& id) const;
+    seastar::lw_shared_ptr<reconcile_history_map> get_or_create_repair_history_for_table(const table_id& id);
+
+    seastar::lw_shared_ptr<reconcile_history_map> get_group0_history() const;
+    seastar::lw_shared_ptr<reconcile_history_map> get_or_create_group0_history();
```

Comment:
"group0_history" makes it easy to confuse with a table of the same name (`system.group0_history`).

Reply:
changed to `get_group0_gc_time`